Some warning during make

Hi.
Ubuntu 22.04 LTS and PHP 8.1
Download your code (zip) and then:

phpize --clean
phpize
./configure
make

During make I see some warning but stats.so is compiled fine and _make test_ reports NO errors :-)
BTW these are the warnings:

libtool: compile:  cc -I. -I/root/php-stat-2.1.0/pecl-math-stats-master -I/root/php-stat-2.1.0/pecl-math-stats-master/include -I/root/php-stat-2.1.0/pecl-math-stats-master/main -I/root/php-stat-2.1.0/pecl-math-stats-master -I/usr/include/php/20210902 -I/usr/include/php/20210902/main -I/usr/include/php/20210902/TSRM -I/usr/include/php/20210902/Zend -I/usr/include/php/20210902/ext -I/usr/include/php/20210902/ext/date/lib -DHAVE_CONFIG_H -g -O2 -DZEND_COMPILE_DL_EXT=1 -c /root/php-stat-2.1.0/pecl-math-stats-master/php_stats.c -MMD -MF php_stats.dep -MT php_stats.lo  -fPIC -DPIC -o .libs/php_stats.o

In file included from /usr/include/php/20210902/Zend/zend.h:33,
                 from /usr/include/php/20210902/main/php.h:32,
                 from /root/php-stat-2.1.0/pecl-math-stats-master/php_stats.c:23:

/root/php-stat-2.1.0/pecl-math-stats-master/php_stats.c: In function ‘zif_stats_stat_percentile’:
/root/php-stat-2.1.0/pecl-math-stats-master/php_stats.c:3212:42: warning: passing argument 3 of ‘zend_hash_sort_ex’ from incompatible pointer type [-Wincompatible-pointer-types]
 3212 |         zend_hash_sort(Z_ARRVAL_P(arg1), stats_array_data_compare, 1);
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~
      |                                          |
      |                                          int (*)(const void *, const void *)

/usr/include/php/20210902/Zend/zend_hash.h:290:42: note: in definition of macro ‘zend_hash_sort’
  290 |         zend_hash_sort_ex(ht, zend_sort, compare_func, renumber)
      |                                          ^~~~~~~~~~~~

/usr/include/php/20210902/Zend/zend_hash.h:286:108: note: expected ‘bucket_compare_func_t’ {aka ‘int (*)(struct _Bucket *, struct _Bucket *)’} but argument is of type ‘int (*)(const void *, const void *)’**
  286 | ZEND_API void  ZEND_FASTCALL zend_hash_sort_ex(HashTable *ht, sort_func_t sort_func, bucket_compare_func_t compare_func, bool renumber);
      |                                                                                      ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~

--- a/php_stats.c
+++ b/php_stats.c
@@ -111,16 +111,18 @@ PHP_MINFO_FUNCTION(stats)
  *
  * This is not correct any more, depends on what compare_func is set to.
  */
+#if PHP_VERSION_ID >= 80000
+static int stats_array_data_compare(Bucket *f, Bucket *s)
+{
+#else
 static int stats_array_data_compare(const void *a, const void *b)
 {
-	Bucket *f;
-	Bucket *s;
+	Bucket *f = (Bucket *) a;
+	Bucket *s = (Bucket *) b;
+#endif
 	int result;
 	zval first;
 	zval second;
-
-	f = (Bucket *) a;
-	s = (Bucket *) b;
 
 	first = f->val;
 	second = s->val;

